src/test/java/ebl/v2/ShippingInstructionIT.java

```diff
@@ -1,29 +1,101 @@
 class ShippingInstructionIT {
 
+  private final String VALID_SHIPPING_INSTRUCTION =
+      loadFileAsString("ValidShippingInstruction.json");
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
   @BeforeAll
   static void configs() throws IOException {
     TestConfig.init();
   }
 
-  // dummy test to verify flow integration works
   @Test
-  void test() {
+  void testValidPostShippingInstruction() {
+    given()
+        .contentType("application/json")
+        .body(VALID_SHIPPING_INSTRUCTION)
+        .post(baseURL() + SHIPPING_INSTRUCTIONS)
+        .then()
+        .assertThat()
+        .body("shippingInstructionID", notNullValue())
+        .body("documentStatus", equalTo("PENC"))
+        .body("shippingInstructionCreatedDateTime", notNullValue())
+        .body("shippingInstructionUpdatedDateTime", notNullValue())
+        .statusCode(HttpStatus.SC_CREATED)
+        .body(jsonSchemaValidator("shippingInstructionRequest"))
+        .extract()
+        .body()
+        .asString();
+  }
+
+  @Test
+  void testFailureOnEmptyBodyForPostShippingInstruction() throws JsonProcessingException {
+
+    ShippingInstructionTO invalid_shipping_instruction = new ShippingInstructionTO();
+
+    given()
+        .contentType("application/json")
+        .body(objectMapper.writeValueAsString(invalid_shipping_instruction))
+        .post(baseURL() + SHIPPING_INSTRUCTIONS)
+        .then()
+        .assertThat()
+        .body("httpMethod", equalTo("POST"))
+        .body("requestUri", containsString("/v2/shipping-instructions"))
+        .body("errors[0].reason", equalTo("invalidInput"))
+        .body("errors[0].message", containsString("isToOrder"))
+        .body("errors[0].message", containsString("shipmentEquipments"))
+        .body("errors[0].message", containsString("isShippedOnboardType"))
+        .body("statusCode", equalTo(400))
+        .body("statusCodeText", equalTo("Bad Request"))
+        .statusCode(HttpStatus.SC_BAD_REQUEST)
+        .extract()
+        .body()
+        .asString();
+  }
+
+  @Test
+  void testFailureOnNonExistingCarrBookingRefPostShippingInstruction() {
+
+    Map<String, Object> map = jsonToMap(VALID_SHIPPING_INSTRUCTION);
+    assert map != null;
+    map.put("carrierBookingReference", UUID.randomUUID().toString());
+
     given()
         .contentType("application/json")
-        .get("/v2/shipping-instructions/" + UUID.randomUUID())
+        .body(map)
+        .post(baseURL() + SHIPPING_INSTRUCTIONS)
         .then()
         .assertThat()
-        .statusCode(200);
+        .body("httpMethod", equalTo("POST"))
+        .body("requestUri", containsString("/v2/shipping-instructions"))
+        .body("errors[0].reason", equalTo("invalidParameter"))
+        .body("errors[0].message", containsString("No shipment found with carrierBookingReference"))
+        .body("statusCode", equalTo(400))
+        .body("statusCodeText", equalTo("Bad Request"))
+        .statusCode(HttpStatus.SC_BAD_REQUEST)
```

Comment:
To be honest, I think we should assert the status code before the content.  If the status code is not 400, then none of the body checks are reliable.

Reply:
Makes sense, I will change it.